src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -62,6 +119,21 @@ public void CurrentProcess_WaitNeverCompletes(int milliseconds)
             Assert.False(Process.GetCurrentProcess().WaitForExit(milliseconds));
         }
 
+        [Theory]
+        [InlineData(0)]  // poll
+        [InlineData(10)] // real timeout
+        public async Task CurrentProcess_WaitAsyncNeverCompletes(int milliseconds)
+        {
+            using (var cts = new CancellationTokenSource(milliseconds))
+            {
+                var token = cts.Token;
+                var process = Process.GetCurrentProcess();
+                var ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => process.WaitForExitAsync(token));
```

Comment:
We do not use `var` if the right-hand side does is not a `new` statement. Makes review easier.

Reply:
Fixed!